ext/soap/php_encoding.c

```diff
@@ -2905,6 +2906,17 @@ static xmlNodePtr to_xml_datetime_ex(encodeTypePtr type, zval *data, char *forma
 		efree(buf);
 	} else if (Z_TYPE_P(data) == IS_STRING) {
 		xmlNodeSetContentLen(xmlParam, BAD_CAST(Z_STRVAL_P(data)), Z_STRLEN_P(data));
+	} else if (Z_TYPE_P(data) == IS_OBJECT) {
+		if (instanceof_function_slow(Z_OBJCE_P(data), php_date_get_interface_ce())) {
+			php_date_obj *dateobj = Z_PHPDATE_P(data);
+			if (dateobj->time) {
+				zend_string *formatted = php_format_date_obj(ext_date_format, ext_date_format_len, dateobj);
```

Comment:
You're testing for `time` here, but IMO that should be done in the new `php_format_date_obj` function, which, if it returns NULL, can then else into your error message branch?

Reply:
Right of course, that makes sense